conformance_tests/tools/metrics/src/test_metric.cpp

```diff
@@ -432,29 +432,30 @@ TEST_F(
 
 class zetMetricQueryTest : public zetMetricGroupTest {
 protected:
-  zet_metric_query_pool_handle_t metricQueryPoolHandle;
-  zet_metric_query_handle_t metricQueryHandle;
+  zet_metric_query_pool_handle_t metricQueryPoolHandle{};
+  zet_metric_query_handle_t metricQueryHandle{};
 
   std::vector<std::string> groupNameList;
-  zet_metric_group_handle_t matchedGroupHandle;
+  zet_metric_group_handle_t matchedGroupHandle{};
   std::string groupName;
-
-  zetMetricQueryTest() {
-
+  void SetUp() override {
     groupNameList = lzt::get_metric_group_name_list(
         device, ZET_METRIC_GROUP_SAMPLING_TYPE_FLAG_EVENT_BASED, false);
+    ASSERT_GT(groupNameList.size(), 0u) << "No query metric groups found";
     groupName = groupNameList[0];
 
     matchedGroupHandle = lzt::find_metric_group(
         device, groupName, ZET_METRIC_GROUP_SAMPLING_TYPE_FLAG_EVENT_BASED);
-    EXPECT_NE(nullptr, matchedGroupHandle);
+    ASSERT_NE(nullptr, matchedGroupHandle);
     metricQueryPoolHandle = lzt::create_metric_query_pool(
         1000, ZET_METRIC_QUERY_POOL_TYPE_PERFORMANCE, matchedGroupHandle);
-    EXPECT_NE(nullptr, metricQueryPoolHandle);
+    ASSERT_NE(nullptr, metricQueryPoolHandle);
     metricQueryHandle = lzt::metric_query_create(metricQueryPoolHandle);
   }
   void TearDown() override {
+    ASSERT_NE(nullptr, matchedGroupHandle);
```

Comment:
could we move to `if` checks.

We have discussions like https://github.com/rigdev/rig/pull/233/files where teardown would be called even if setup fails.

Reply:
Resolved, Thank you.